### PR TITLE
[DDO-2618] Delete 2k lines of code

### PR DESCRIPTION
**Pull request by jack-r-warren**
It doesn't look any different when you load it into a browser. There's some slight changes, basically some standardization. List sort order is more stable now. There's 2k fewer lines of code and it uses the new v2_meta, v2_errorBoundary, and v2_routeConvention features.

v2_meta is just an alternative way of setting `<meta>` tags. It's better than the v1 version, though we weren't having issues with it--was good just to start using the better mechansim.

v2_errorBoundary means that each route just has a single ErrorBoundary export--no need for both an ErrorBoundary (for thrown `Error` types) and a CatchBoundary (for thrown `Response` types). The boundary just handles both. Removes some code duplication. I also took the opportunity to simplify a bit of how Sherlock's errors work to agree more with type inference.

[v2_routeConvention means flat file names](https://github.com/remix-run/remix/discussions/4482), instead of deeply-nested folders. I'm not actually sure which I like better. The file names are, well, long.

> We're in a unique case because the we actually have a lot of duplication. `**/chart-releases/*` is _mostly_ duplicated three times; every `**/new` or `**/add` page reuses fields from `**/edit`. Basically, routes here often share components. This means that the fun co-location that flat routes bring isn't super useful to us, unless we decide we're cool importing stuff basically from one route to another (which isn't an anti-pattern in remix; it's how you're supposed to do useOutletContext to get parent data)

Basically I kinda wanna just live with it for a bit. The nice thing is that stuff is way more abstracted and in separate files now, so if we wanted to go to the colocation-style flat routes we could do that just by shuffling files--wouldn't need to edit by hand like I had to do here.

Potentially the most significant thing is me getting way better at typescript. I removed almost every single type assertion from Beehive by adopting [better patterns that the Remix folks have been working on](https://github.com/remix-run/remix/releases/tag/remix%401.7.4)--basically the types from the Sherlock Typescript client flow all the way from the server-side loader/action into the page component itself (which means the types flow across the wire, which is neat).

### CODE REVIEW

**Review by mflinn-broad — APPROVED** [review]
:rubberstamp. Will look more at some point just because I'm curious
